frontend/src/util/toolbar_functions.js

```diff
@@ -1,5 +1,42 @@
-export function print() {
+import jsPDF from 'jspdf';
+import { getRedactionsOnTextNode } from "@/util/editorRedactionUtils";
+import { ACCEPTED_PREFIX } from "../util/editorRedactionUtils";
 
+export const toPDF = (body) => {
+    return Array.from(body.document.documentBody)
+        .map((paragraph) => {
+            const p_tag_start = "<p>";
+            const p_tag_end = "</p>";
+            const internal_text = paragraph.children.map((node) => {
+                const isAccepted = getRedactionsOnTextNode(node, ACCEPTED_PREFIX).size > 0;
+                let text = node.text.replace("\n", p_tag_end + "<br>" + p_tag_start)
+                return isAccepted ? "XXXXX" : text;
+            });
+
+            const res = p_tag_start + internal_text.join(" ") + p_tag_end;
+            return res;
+        }
+    )
+    .join("<br>");
+};
+
+export function exportDoc(document) {
+    const pdf_doc = new jsPDF({
+        orientation: "portrait",
+        unit: "pt",
+        format: 'letter'
+    });
+
+    let str = toPDF(document)
+
+    pdf_doc.html(str, {
+        async callback(doc) {
+            await doc.save('pdf_name');
+        },
+        width: 600,
+        windowWidth: 600,
+        margin: [10, 10, 10, 10]
```

Comment:
also sorry nit but larger margins pls, let's make it like 30? so that it looks like a normal document's margins. Like try to make it look like when you print Google Docs